master/buildbot/db/model.py

```diff
@@ -104,6 +104,7 @@ class Model(base.DBConnectorComponent):
     # claim is made by the master referenced by masterid.
     buildrequest_claims = sautils.Table(
         'buildrequest_claims', metadata,
+        sa.Column('dummy', sa.Integer, primary_key=True, autoincrement=True),
         sa.Column('brid', sa.Integer, sa.ForeignKey('buildrequests.id'),
```

Comment:
If you still want to keep having a buildrequest_claims, then the masterid column shouldn't be null. First of all because the masterid colum is a foreign key that references a primary key in the buildrequest table, and primary keys cannot be null, so it will never reference a null value. And second because the combination of brid and master id are both unique and identify a claim, so they should be a composite primary key.

When a master inserts a row in buildrequest_claims, it inserts the brid and the masterid both at the same time, so masterid already has a non-null value.

Do you prefer this approach for my next spin of this pull request?